<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PluginGame5GenerateSituationDialog</class>
 <widget class="QDialog" name="PluginGame5GenerateSituationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>667</width>
    <height>537</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Игра 5</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="3">
    <widget class="QWidget" name="hiddenWidget" native="true">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="2" column="0">
       <widget class="QGroupBox" name="heuristicsGroupBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="title">
         <string>Эвристика</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="2" rowspan="4">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="3" column="0">
          <widget class="QRadioButton" name="radioButtonCustom">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QRadioButton" name="radioButtonDistance">
           <property name="text">
            <string>Растояния_фишек_до_места</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QRadioButton" name="radioButton0">
           <property name="text">
            <string>0 - Поиск в ширину</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QRadioButton" name="radioButtonQuantity">
           <property name="text">
            <string>Кол_во_фишек_не_на_месте</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="lineEditCustom"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QGroupBox" name="compareTopsGroupBox">
        <property name="title">
         <string>Прохождение графа</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="0">
          <widget class="QCheckBox" name="checkBoxCopareTop">
           <property name="text">
            <string>Сравнивать вершину с раскрытыми</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QGroupBox" name="costSetupGroupBox">
        <property name="title">
         <string>Стоимость принятия правил</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0">
          <widget class="QLabel" name="headerLabelDirection">
           <property name="text">
            <string>Движение</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="headerLabelCalc">
           <property name="text">
            <string>Вычисление</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="moveRightLabel">
           <property name="text">
            <string>Вправо</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="moveRightCalcSwitcher">
           <item>
            <property name="text">
             <string>after</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>before</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="moveLeftLabel">
           <property name="text">
            <string>Влево</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="moveLeftCalcSwitcher">
           <item>
            <property name="text">
             <string>after</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>before</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="moveUpLabel">
           <property name="text">
            <string>Вверх</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="moveUpCalcSwitcher">
           <item>
            <property name="text">
             <string>after</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>before</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="moveDownLabel">
           <property name="text">
            <string>Вниз</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="moveDownCalcSwitcher">
           <item>
            <property name="text">
             <string>after</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>before</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="headerLabelCost">
           <property name="text">
            <string>Стоимость</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="CheckableLineEdit" name="moveRightCost" native="true"/>
         </item>
         <item row="2" column="2">
          <widget class="CheckableLineEdit" name="moveLeftCost" native="true"/>
         </item>
         <item row="3" column="2">
          <widget class="CheckableLineEdit" name="moveUpCost" native="true"/>
         </item>
         <item row="4" column="2">
          <widget class="CheckableLineEdit" name="moveDownCost" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="5" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QWidget" name="mainWidget" native="true">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="2">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="buttonRandomLineup">
        <property name="text">
         <string>Перемешать</string>
        </property>
        <property name="autoDefault">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="buttonRightLineup">
        <property name="text">
         <string>По порядку</string>
        </property>
        <property name="autoDefault">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="solvabilityCheck">
        <property name="text">
         <string>Только решаемые</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="buttonHide">
        <property name="maximumSize">
         <size>
          <width>40</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
        <property name="autoDefault">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="3" colspan="2">
       <widget class="QPushButton" name="buttonOk">
        <property name="text">
         <string>Ok</string>
        </property>
        <property name="autoDefault">
         <bool>true</bool>
        </property>
        <property name="default">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="3">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="buttonSetLineup">
        <property name="text">
         <string>Задать..</string>
        </property>
        <property name="autoDefault">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="5">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QGridLayout" name="gridLayout_7">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="1" column="1">
          <widget class="Board" name="gameBoard" native="true"/>
         </item>
         <item row="2" column="1">
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="2">
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Board</class>
   <extends>QWidget</extends>
   <header>app/rdo_studio/plugins/game5/src/board.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CheckableLineEdit</class>
   <extends>QWidget</extends>
   <header>app/rdo_studio/plugins/game5/src/checkable_line_edit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>buttonOk</tabstop>
  <tabstop>buttonRandomLineup</tabstop>
  <tabstop>solvabilityCheck</tabstop>
  <tabstop>buttonRightLineup</tabstop>
  <tabstop>buttonSetLineup</tabstop>
  <tabstop>buttonHide</tabstop>
  <tabstop>radioButton0</tabstop>
  <tabstop>radioButtonQuantity</tabstop>
  <tabstop>radioButtonDistance</tabstop>
  <tabstop>radioButtonCustom</tabstop>
  <tabstop>lineEditCustom</tabstop>
  <tabstop>checkBoxCopareTop</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
